src/compiler/scala/tools/nsc/typechecker/Infer.scala

```diff
@@ -148,7 +148,7 @@ trait Infer extends Checkable {
   }
 
   def skipImplicit(tp: Type) = tp match {
-    case mt: MethodType if mt.isImplicit  => mt.resultType
+    case mt: MethodType if mt.isImplicit  => mt.resultType //
```

Comment:
- [ ] This looks like a potential bug with more than one implicit param list

`skipImplicit` is called from this error message:

```
scala> def foo[A](implicit a: DummyImplicit): Array[A] = null
foo: [A](implicit a: DummyImplicit)Array[A]

scala> foo: Seq[_]
<console>:18: error: polymorphic expression cannot be instantiated to expected type;
 found   : [A]Array[A]
 required: Seq[_]
       foo: Seq[_]
       ^

scala> def foo[A](implicit a: DummyImplicit)(implicit b: DummyImplicit): Array[A] = null
foo: [A](implicit a: DummyImplicit)(implicit b: DummyImplicit)Array[A]

scala> foo: Seq[_]
<console>:18: error: polymorphic expression cannot be instantiated to expected type;
 found   : [A](implicit b: DummyImplicit)Array[A]
 required: Seq[_]
       foo: Seq[_]
       ^
```

It is also called from `Infer::inferForApproxPt` as follows (in the context of inferring type paramsters for constructor patterns for case classes:

```
val resTpInst  = skipImplicit(ctorTpInst.finalResultType)
```

However, AFAICT the call to `finalResultType` would peel off all `MethodTypes` rendering `skipImplicit` a redundant no-op.